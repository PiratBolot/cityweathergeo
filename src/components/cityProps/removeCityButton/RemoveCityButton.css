.remove_city_button {
    display: inline-block;
    background-color: #435257;
    color: white;
    margin: 10px;
    padding: 10px;
    border: none;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    box-shadow: 0 3px 20px rgba(0, 0, 0, .25);
}