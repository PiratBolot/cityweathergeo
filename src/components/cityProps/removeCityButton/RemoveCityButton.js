import React from 'react'
import './RemoveCityButton.css';
import {deleteTrackedCity} from "../../../actions/Actions";
import {connect} from "react-redux";

class RemoveCityButton extends React.Component {
    constructor(props) {
        super(props);
        this.removeTrackedCity = this.removeTrackedCity.bind(this);
    }

    async removeTrackedCity(e) {
        e.preventDefault();
        this.props.deleteCity(this.props.city);
    };

    render() {
        return (
            <button className="remove_city_button" onClick={this.removeTrackedCity}>X</button>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    deleteCity: (city) => dispatch(deleteTrackedCity(city))
});

export default connect(null, mapDispatchToProps)(RemoveCityButton);