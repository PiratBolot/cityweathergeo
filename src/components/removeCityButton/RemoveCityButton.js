import React from 'react'
import './RemoveCityButton.css';
import {deleteTrackedCity} from "../../actions/Actions";
import {connect} from "react-redux";

class RemoveCityButton extends React.Component {
    constructor(props) {
        super(props);
        this.removeTrackedCity = this.removeTrackedCity.bind(this);
    }

    removeTrackedCity = async (e) => {
        e.preventDefault();
        this.props.deleteCity(this.props.city);
    };

    render = () => (
        <button className="remove_city_button" onClick={this.removeTrackedCity}>X</button>
    )
}

const mapDispatchToProps = (dispatch) => ({
    deleteCity: (city) => dispatch(deleteTrackedCity(city))
});

export default connect(null, mapDispatchToProps)(RemoveCityButton);