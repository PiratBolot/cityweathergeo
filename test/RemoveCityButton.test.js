import React from 'react';
import {mount} from '../enzyme/enzyme';
import {applyMiddleware, createStore} from "redux";
import {Provider} from "react-redux";
import rootReducer from "../src/reducers/Reducers";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";
import RemoveCityButton from "../src/components/cityProps/removeCityButton/RemoveCityButton";

describe('Remove City Button', () => {
    const city = "Syktyvkar";
    const initialState = {
        trackedCities: ['Syktyvkar', 'Moscow', 'Kiev'],
        errorMessage: {isError: false, errorMsg: ""},
        geolocation: {}
    };
    let store,wrapper;
    beforeEach(()=>{
        store = createStore(rootReducer, initialState, applyMiddleware(thunk));
        wrapper = mount( <Provider store={store}><RemoveCityButton city={city} /></Provider> );
    });
    it('+++capturing Snapshot of RemoveCityButton', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});