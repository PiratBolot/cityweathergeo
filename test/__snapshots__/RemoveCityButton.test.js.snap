// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove City Button +++capturing Snapshot of RemoveCityButton 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(RemoveCityButton)
    city="Syktyvkar"
  >
    <RemoveCityButton
      city="Syktyvkar"
      deleteCity={[Function]}
    >
      <button
        className="remove_city_button"
        onClick={[Function]}
      >
        X
      </button>
    </RemoveCityButton>
  </Connect(RemoveCityButton)>
</Provider>
`;
